src/Order/OrderTests/BookingConcurrencyTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using OrderAPI.DTOs;
+using Refit;
+using System.Net;
+
+namespace OrderTests
+{
+	public class BookingConcurrencyTests : IClassFixture<WebApplicationFactory<Program>>
+	{
+		private readonly WebApplicationFactory<Program> _factory;
+
+		public BookingConcurrencyTests(WebApplicationFactory<Program> factory)
+		{
+			// to avoid conflicts with local docker
+			_factory = factory.WithWebHostBuilder(builder => builder.UseEnvironment("QA"));
+		}
+
+		[Theory]
+		[InlineData("pessimistic")]
+		[InlineData("optimistic")]
+		public async Task CartController_Book_OneSuccessfulBooking(string url)
+		{
+			await new EndToEndExecutioner().Execute(async () =>
+			{
+				var apiClient = RestService.For<ICartApi>(_factory.CreateClient());
+				List<Guid> guids = [];
+				foreach (var _ in Enumerable.Range(0, 1000))
+				{
+					var guid = Guid.NewGuid();
+					var cartItemModels = Enumerable.Range(1, 4)
+						.Select(index => new CartItemInputModel() { EventId = 1, SeatId = index, UserId = 1 })
+						.ToArray();
+					foreach (var cartItem in cartItemModels)
+						await apiClient.AddItemToCart(cartItem, guid);
+					guids.Add(guid);
+				}
+
+				List<HttpResponseMessage> responses = [];
+				await Parallel.ForEachAsync(guids, async (guid, _) =>
+				{
+					responses.Add(await apiClient.Book(guid, url));
+				});
```

Comment:
With this approach, you should better use concurrent collection like ConcurrentBag<HttpResponseMessage>
Also, you could try something like this:
```
var tasks = guids.Select(guid => apiClient.Book(guid, url));
var responses = await Task.WhenAll(tasks);
```